### PR TITLE
Remove skips from test bodies

### LINKED ISSUE #29: Remove `skip` lines from test runner code bodies
Given:
exercism_test_results

Tidy up the output by excluding the "skip" lines.

### PR DESCRIPTION
**Pull request by glennj**
Resolves #29

### CODE REVIEW

File: bin/run.sh [hunk 1]
```diff
@@ -35,32 +35,6 @@ main() {
     build_report "$output_file" "$json_result_file"
 }
 
-to_json_value() {
```
Comment (glennj) [inline bin/run.sh:38]: This function just got moved lower down.

File: bin/run.sh [hunk 2]
```diff
@@ -112,6 +86,12 @@ get_test_bodies() {
                 if [[ $line =~ $end_test_re ]]; then
                     test_bodies["$name"]=$(printf '%s\n' "${body[@]}")
                     state="out"
+                elif [[ $line == *BATS_RUN_SKIPPED*skip* ]]; then
+                    # skip the skips
+                    continue
+                elif ((${#body[@]} == 0)) && [[ $line == *([[:blank:]]) ]]; then
+                    # ignore blank lines at the top of the test body
+                    continue
```
Comment (glennj) [inline bin/run.sh:94]: This is the important new code.

File: tests/data/last_fails/expected_results.json [hunk 3]
```diff
@@ -1,4 +1,4 @@
 { "version": 2, "status": "fail", "tests": [
   { "name": "say one", "status": "pass", "test_code": "run bash last_fails.sh one\n[ \"$status\" -eq 0 ]\n[ \"$output\" == \"one\" ]"},
   { "name": "say two", "status": "pass", "test_code": "run bash last_fails.sh two\n[ \"$status\" -eq 0 ]\n[ \"$output\" == \"two\" ]"},
-  { "name": "say three", "status": "fail", "test_code": "run bash last_fails.sh three\n[ \"$status\" -eq 0 ]\n[ \"$output\" == \"three\" ]", "message": "(in test file last_fails_test.sh, line 18)\n  `[ \"$output\" == \"three\" ]' failed\n"}]}
\ No newline at end of file
+  { "name": "say three", "status": "fail", "test_code": "run bash last_fails.sh three\n[ \"$status\" -eq 0 ]\n[ \"$output\" == \"three\" ]", "message": "(in test file last_fails_test.sh, line 26)\n  `[ \"$output\" == \"three\" ]' failed\n"}]}
```
Comment (glennj) [inline tests/data/last_fails/expected_results.json:4]: line number changed

File: tests/data/middle_fails/expected_results.json [hunk 4]
```diff
@@ -1,4 +1,4 @@
 { "version": 2, "status": "fail", "tests": [
   { "name": "say one", "status": "pass", "test_code": "run bash middle_fails.sh one\n[ \"$status\" -eq 0 ]\n[ \"$output\" == \"one\" ]"},
-  { "name": "say two", "status": "fail", "test_code": "run bash middle_fails.sh two\n[ \"$status\" -eq 0 ]\n[ \"$output\" == \"two\" ]", "message": "(in test file middle_fails.bats, line 14)\n  `[ \"$output\" == \"two\" ]' failed\n"},
-  { "name": "say three", "status": "pass", "test_code": "run bash middle_fails.sh three\n[ \"$status\" -eq 0 ]\n[ \"$output\" == \"three\" ]"}]}
\ No newline at end of file
+  { "name": "say two", "status": "fail", "test_code": "run bash middle_fails.sh two\n[ \"$status\" -eq 0 ]\n[ \"$output\" == \"two\" ]", "message": "(in test file middle_fails.bats, line 18)\n  `[ \"$output\" == \"two\" ]' failed\n"},
```
Comment (glennj) [inline tests/data/middle_fails/expected_results.json:3]: line number changed.

File: bin/run.sh (hunk 2 above)
Comment (IsaacG) [reply bin/run.sh:94]: I'm on my phone so ... do you explicitly set extglob in this script?

**Review by IsaacG — APPROVED** [review]

File: bin/run.sh (hunk 2 above)
Comment (glennj) [reply bin/run.sh:94]: extglob is [enabled automatically inside `[[...]]`](https://www.gnu.org/software/bash/manual/bash.html#index-_005b_005b)

> When the ‘==’ and ‘!=’ operators are used, the string to the right of the operator is considered a pattern and matched according to the rules described below in Pattern Matching, as if the extglob shell option were enabled.

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**